### PR TITLE
Leader election config update.

### PR DESCRIPTION
**Pull request by cappetersson**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[1391](https://github.com/openshift/hypershift/issues/1391)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kcm/deployment.go [hunk 1]
```diff
@@ -348,7 +350,8 @@ func kcmArgs(p *KubeControllerManagerParams) []string {
 		"--kube-api-qps=150",
 		"--leader-elect-resource-lock=configmapsleases",
 		"--leader-elect=true",
-		"--leader-elect-retry-period=3s",
+		fmt.Sprintf("--leader-elect-renew-deadline duration=%s", config.KCMRecommendedRenewDeadline),
```
Comment (muraee) [inline control-plane-operator/controllers/hostedcontrolplane/kcm/deployment.go:353]:
```suggestion
		fmt.Sprintf("--leader-elect-renew-deadline-duration=%s", config.KCMRecommendedRenewDeadline),
```
I believe you are missing a `-` here

Comment (cappetersson) [reply control-plane-operator/controllers/hostedcontrolplane/kcm/deployment.go]: Nice catch. Thank you for the review!

**Comment by cappetersson** [comment]
@muraee 
I noticed that the `ci/prow/e2e-aws`-test is failing on other open PRs. Would this indicate that the error is not specific to my changes? Just checking to see if there are additional things I need to do to get this PR merged.

**Comment by muraee** [comment]
@cappetersson yeah, I think there was another problem last week.
/retest-required

**Comment by muraee** [comment]
@cappetersson we don't a direct way to check the status of our CI, we have a blocking job on CI stream so you could check that release for an indication (e.g. e2e-hypershift is failing here https://amd64.ocp.releases.ci.openshift.org/releasestream/4.14.0-0.ci/release/4.14.0-0.ci-2023-05-29-185629).

Actually we have a problem with our CI currently :)

**Comment by cappetersson** [comment]
Oh that is super helpful, thanks for showing me that @muraee! I will keep my eyes on the release page and then re-run the tests when I can see that those problems have been solved:+1:

**Comment by muraee** [comment]
@cappetersson we need `ci/prow/e2e-aws` which is our main e2e test

**Comment by cappetersson** [comment]
@muraee
I've been keeping an eye on other PRs that previously failed on the same test that my PR is failing on and now I'm noticing that they are passing. I've been running a couple of retests today but they are still failing. Is there something specific I need to do since I do not think it is because of my code edits based on the PR tests failing on the same stage as for other PRs.

**Comment by muraee** [comment]
@cappetersson I think rebasing your branch should help.

**Comment by cappetersson** [comment]
Ah yes there it is. How were you able to know to look in that specific folder for the logs? Is that just something you know or would I be able to tell based on the information given by the failing tests.

**Comment by cappetersson** [comment]
@muraee 

Finally was able to get all tests to pass. Thanks for all the help on this.

**Comment by cappetersson** [comment]
@muraee 
Any idea on when I will be able to get some reviews on this?
Thanks

**Pull request merged**